client/http/client.go

```diff
@@ -339,508 +230,178 @@ func (c *client) request(
-		"SetScheduleConfig", ScheduleConfig,
-		http.MethodPost, configJSON, nil)
-}
-
-// GetStores gets the stores info.
-func (c *client) GetStores(ctx context.Context) (*StoresInfo, error) {
-	var stores StoresInfo
-	err := c.requestWithRetry(ctx,
-		"GetStores", Stores,
-		http.MethodGet, nil, &stores)
-	if err != nil {
-		return nil, err
-	}
-	return &stores, nil
-}
-
-// GetStore gets the store info by ID.
-func (c *client) GetStore(ctx context.Context, storeID uint64) (*StoreInfo, error) {
-	var store StoreInfo
-	err := c.requestWithRetry(ctx,
-		"GetStore", StoreByID(storeID),
-		http.MethodGet, nil, &store)
-	if err != nil {
-		return nil, err
-	}
-	return &store, nil
-}
-
-// GetClusterVersion gets the cluster version.
-func (c *client) GetClusterVersion(ctx context.Context) (string, error) {
-	var version string
-	err := c.requestWithRetry(ctx,
-		"GetClusterVersion", ClusterVersion,
-		http.MethodGet, nil, &version)
-	if err != nil {
-		return "", err
-	}
-	return version, nil
-}
-
-// GetAllPlacementRuleBundles gets all placement rules bundles.
-func (c *client) GetAllPlacementRuleBundles(ctx context.Context) ([]*GroupBundle, error) {
-	var bundles []*GroupBundle
-	err := c.requestWithRetry(ctx,
-		"GetPlacementRuleBundle", PlacementRuleBundle,
-		http.MethodGet, nil, &bundles)
-	if err != nil {
-		return nil, err
-	}
-	return bundles, nil
-}
-
-// GetPlacementRuleBundleByGroup gets the placement rules bundle by group.
-func (c *client) GetPlacementRuleBundleByGroup(ctx context.Context, group string) (*GroupBundle, error) {
-	var bundle GroupBundle
-	err := c.requestWithRetry(ctx,
-		"GetPlacementRuleBundleByGroup", PlacementRuleBundleByGroup(group),
-		http.MethodGet, nil, &bundle)
-	if err != nil {
-		return nil, err
-	}
-	return &bundle, nil
-}
-
-// GetPlacementRulesByGroup gets the placement rules by group.
-func (c *client) GetPlacementRulesByGroup(ctx context.Context, group string) ([]*Rule, error) {
-	var rules []*Rule
-	err := c.requestWithRetry(ctx,
-		"GetPlacementRulesByGroup", PlacementRulesByGroup(group),
-		http.MethodGet, nil, &rules)
-	if err != nil {
-		return nil, err
+	for _, member := range membersInfo.Members {
+		if membersInfo.Leader != nil && member.GetMemberId() == membersInfo.Leader.GetMemberId() {
+			newLeaderAddrIdx = len(newPDAddrs)
+		}
+		newPDAddrs = append(newPDAddrs, member.GetClientUrls()...)
 	}
-	return rules, nil
-}
-
-// SetPlacementRule sets the placement rule.
-func (c *client) SetPlacementRule(ctx context.Context, rule *Rule) error {
-	ruleJSON, err := json.Marshal(rule)
-	if err != nil {
-		return errors.Trace(err)
+	// Prevent setting empty addresses.
+	if len(newPDAddrs) == 0 {
+		log.Error("[pd] http client get empty member addresses")
+		return
 	}
-	return c.requestWithRetry(ctx,
-		"SetPlacementRule", PlacementRule,
-		http.MethodPost, ruleJSON, nil)
+	ci.setPDAddrs(newPDAddrs)
+	ci.setLeaderAddrIdx(newLeaderAddrIdx)
+	log.Info("[pd] http client update members info",
```

Comment:
Should this log level be info? will print every minute

Reply:
I added a condition to ensure that it only prints when members change. PTAL.

Reply:
<img width="2035" alt="image" src="https://github.com/tikv/pd/assets/1446531/05f4204a-f2fe-47ae-bd7e-0420c984ee36">